### PR TITLE
dhcp: include ddns-hostname for static host entries

### PR DESCRIPTION
**Pull request by rfc4711**
Proposed code changs per issue #4452

### CODE REVIEW

**Comment by AdSchellevis** [comment]
same https://github.com/opnsense/core/pull/5288 ?

**Comment by rfc4711** [comment]
@fichtner No, I originally opened ticket #4452 for a bug, since my DNS is running on a separate VM and the host-names are not updated to the names I have configured on opnsense dhcp config, but to the device originated one. First time submitting a patch for me, when I looked at the code it didn't seem too difficult adding the two lines.

**Comment by rfc4711** [comment]
Hi Franco,

Apologies, didn't read your last email carefully. The changes are good to
commit as they are for the host-name changes.

I need to check on the php part on how to better configure the
ddns-zone name per host. Though this should be another enhancement request.

Best regards,
Sascha

**Comment by rfc4711** [comment]
@fichtner how will this conflict be resolved? another commit removed the ddns configurations for v6. why does it take that long to merge this small change?

**Comment by fichtner** [comment]
@rfc4711 I have no easy way to confirm this... yes this is a small change but not a trivial one in a non-critical system.. What we found over the last month is that IPv6 host name registration doesn't work at all in ISC-DHCP. That's why we are backing out changes that are not working.

**Comment by rfc4711** [comment]
@fichtner, I believe the setting _ddns-hostname_ should suffice to specify a ddns hostname for v4 and v6. However, if the whole v6 code needs a re-write, can you remove my v6 modifications? Except for setting the ddns hostname, I only changed some whitespace formatting of the config file.

**Comment by rfc4711** [comment]
@fichtner, can you please help me update the master branch on my checkout? I created a pull request to update my local master on GitHub to edit my patch, but the pull request failed.

**Comment by rfc4711** [comment]
@fichtner this change should be ready now to be committed to master. All conflicts seem resolved.

File: src/etc/inc/plugins.inc.d/dhcpd.inc [hunk 1]
```diff
@@ -1627,6 +1630,12 @@ EOD;
                     $dhhostname = str_replace(" ", "_", $sm['hostname']);
                     $dhhostname = str_replace(".", "_", $dhhostname);
                     $dhcpdv6conf .= "  option host-name {$dhhostname};\n";
+                  
+                    if ($need_ddns_updates) {
+                        $dhcpdv6conf .= "  ddns-hostname {$dhhostname};\n";
+                    }
+                    $dhcpdv6conf .= "  set hostname-override = config-option host-name;\n";
+
```
Comment (fichtner) [inline src/etc/inc/plugins.inc.d/dhcpd.inc:1638]:
```suggestion
```

mismerge

File: src/etc/inc/plugins.inc.d/dhcpd.inc [hunk 2]
```diff
@@ -1246,8 +1247,10 @@ EOD;
     }
 
     if ($need_ddns_updates) {
+        $dhcpdconf .= "\n";
         $dhcpdconf .= "ddns-update-style interim;\n";
         $dhcpdconf .= "update-static-leases on;\n";
+        $dhcpdconf .= "\n";
```
Comment (fichtner) [inline src/etc/inc/plugins.inc.d/dhcpd.inc:1253]:
```suggestion
        $dhcpdconf .= "\nddns-update-style interim;\n";
        $dhcpdconf .= "update-static-leases on;\n";
```

condensed to ipv6 style

**Comment by fichtner** [comment]
Merging this manually for #4452

**Pull request closed without merging**